oshi-core/src/test/java/oshi/util/MemoizerTest.java

```diff
@@ -115,8 +115,7 @@ public Long get() {
                 boolean guaranteedIteration = false;
                 long now;
                 while ((now = System.nanoTime()) < beginNanos + iterationDurationNanos
-                        || now < firstSupplierCallNanos + ttlNanos
-                        || (guaranteedIteration = !guaranteedIteration)) {
+                        || now < firstSupplierCallNanos + ttlNanos || (guaranteedIteration = !guaranteedIteration)) {
```

Comment:
Oh! How did we get from `now - firstSupplierCallNanos < ttlNanos` to `now < firstSupplierCallNanos + ttlNanos`?
It's not the same thing when we are talking about `System.nanoTime`. The results of this method are allowed to overflow, therefore we can only compare deltas, but we can't compare absolute values (see the javadoc of the method).

So `now < firstSupplierCallNanos + ttlNanos` is not the same as `now - firstSupplierCallNanos < ttlNanos. And we want the latter. The same applies to `(now = System.nanoTime()) < beginNanos + iterationDurationNanos`. Somehow I missed this previously.

Reply:
I changed that for readability.  Didn't think about the potential for overflow.

Reply:
Notice also how expirationNanos is compared with now in Memoizer.memoize:
```
long nanos = expirationNanos;
now - nanos >= 0;
```
A delta is compared with 0 rather than just comparing `now >= nanos`. That's not just for fun :)

Reply:
Well, the only concievable case where A - B < C gives a different result than A < B + C is if "B+C" overflows.  But given that we are assured that A will exceed the value of B+C during this program execution (which would invalidate the interval method) I maintain the original calculation is still valid.   Unless expirationNanos is on the order of hundreds of years and the test runs forever. :)

Reply:
> the only conceivable case where A - B > C gives a different result than A > B + C is if "B+C" overflows

No, consider the following:
```java
long a = Long.MIN_VALUE;
long b = (Long.MAX_VALUE - 1);
long c = 1;
System.out.println("(a - b > c) is " + (a - b > c));//true
System.out.println("(a > b + c) is " + (a > b + c));//false
```
`b + c` does not overflow, and yet the results are different. The thing is that `Long.MIN_VALUE < Long.MAX_VALUE`. But if you received these values from `System.nanoTime` in the order `Long.MAX_VALUE`, `Long.MIN_VALUE` it means that you did two invocations of this method with an interval of about 1ns between them, i.e. `Long.MIN_VALUE`ns is greater than `Long.MAX_VALUE`ns when we are talking about `System.nanoTime`, and `Long.MIN_VALUE - Long.MAX_VALUE > 0` is `true`.

Reply:
And when you start a JVM, `System.nanoTime` does not return values starting from 0: _The value returned represents nanoseconds since some fixed but arbitrary origin time (perhaps in the future, so values may be negative)_; it can easily start from `Long.MAX_VALUE` or from any other valid long.

Reply:
OK... :)